Sources/NIOPosix/SelectableEventLoop.swift

```diff
@@ -917,3 +929,33 @@ internal func assertExpression(_ body: () -> Bool) {
         }()
     )
 }
+
+extension SelectableEventLoop {
+    @inlinable
+    func scheduleCallback(
+        at deadline: NIODeadline,
+        handler: some NIOScheduledCallbackHandler
+    ) throws -> NIOScheduledCallback {
+        let taskID = self.scheduledTaskCounter.loadThenWrappingIncrement(ordering: .relaxed)
```

Comment:
Something for the future potentially. With this new fast implementation to schedule callbacks it might be worth looking at the normal `scheduleTask` APIs again to see if we can revamp their internal implementation to avoid some of the allocations that they currently have. We can't get rid of all of them i.e. the returned ELP/ELF but potentially the internal allocs where we close over state.

Reply:
Sure, I think @Lukasa's tome comment here will provide good content for someone wanting to see if we can improve it: https://github.com/apple/swift-nio/pull/2759#issuecomment-2252500183.

But as to whether we _should_, it sounded like the back and forth on the motivation was that the older APIs are a little confusing to hold. Without wanting to open another can of worms, I guess a bigger question is: are there real use cases for `scheduleTask` that cannot be implemented with `scheduleCallback`? If the answer is, _no_, then should we consider deprecating the old one given the discussion about its confusing shape?